doc/contributing/adding-v8-fast-api.md

```diff
@@ -1,91 +1,397 @@
+                    v8::Local<v8::Value> argument,
+                    const v8::FastApiCallbackOptions& options) {
+  if (!argument->IsObject()) {
+    return false;
+  }
 
-Passing `true` to the `fallback` option will force V8 to run the slow path
-with the same arguments.
+  // In order to create any Local handles, we first need a HandleScope
+  v8::HandleScope HandleScope(options.isolate);
 
-In V8, the options fallback is defined as `FastApiCallbackOptions` inside
-[`v8-fast-api-calls.h`](../../deps/v8/include/v8-fast-api-calls.h).
+  v8::Local<v8::Object> object = argument.As<v8::Object>();
+  v8::Local<v8::Value> value;
+  if (!object->Get(options.isolate->GetCurrentContext(),
+                   v8::Symbol::GetIterator(options.isolate)).ToLocal(&value)) {
+    return false;
+  }
+  return value->IsFunction();
+}
+```
 
-* C++ land
+The same applies if the fast callback calls other functions which themselves
+create `v8::Local` handles, unless those functions create their own
+`v8::HandleScope`. In general, if the fast callback interacts with
+`v8::Local` handles within the body of the callback, it likely needs a handle
+scope.
 
-  Example of a conditional fast path on C++
+## Debug tracking of Fast API callbacks
 
-  ```cpp
-  // Anywhere in the execution flow, you can set fallback and stop the execution.
-  static double divide(const int32_t a,
-                       const int32_t b,
-                       v8::FastApiCallbackOptions& options) {
-    if (b == 0) {
-      options.fallback = true;
-      return 0;
-    } else {
-      return a / b;
-    }
+In order to allow the test suite to track when a function call uses the Fast
+API path, add the `TRACK_V8_FAST_API_CALL` macro to your fast callback.
+
+```cpp
+bool FastIsEven(const int32_t n) {
+  TRACK_V8_FAST_API_CALL("util.isEven");
+  return n % 2 == 0;
+}
+```
+
+The tracking key must be unique, and should be of the form:
+
+`<namespace> "." <function> [ "." <subpath> ]`
+
+The above example assumes that the fast callback is bound to the `isEven`
+method of the `util` module binding. To track specific subpaths within the
+callback, use a key with a subpath specifier, like `"util.isEven.error"`.
+
+These tracking events can be observed in debug mode, and are used to test that
+the fast path is being correctly invoked. See
+[Testing Fast API callbacks](#testing-fast-api-callbacks) for details.
+
+## Handling errors
+
+It is now possible to throw errors from within the Fast API.
+
+Any fast callback that might potentially need to throw an error back to the
+JavaScript environment should accept a final `options` argument of type
+`const v8::FastApiCallbackOptions&`. V8 will pass the isolate pointer in
+`options.isolate`.
+
+The callback should then throw a JavaScript error in the standard fashion. It
+also needs to return a dummy value, to satisfy the function signature.
+
+As above, initializing a `v8::HandleScope` is mandatory before any operations
+which create local handles.
```

Comment:
As with https://github.com/nodejs/node/pull/58999#discussion_r2193624588, I'm fairly sure that invoking the ERROR macros (which build the error within a local object handle) necessitates a HandleScope here, but someone else may have a definitive answer?